From b48fcb0bff14044caece520611233feb157401bf Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Mon, 24 Jun 2024 14:56:37 +0200
Subject: [PATCH 17/21] gdbserver_tests filters remove python rpm module load
 warning

gdb on Fedora will warn not being able to load the rpm python module.
Unable to load 'rpm' module.  Please install the python3-rpm package.
Filter out that message so tests don't fail.

(cherry picked from commit d6c7d47eb4c13a3d83c091d453f527a1b74cf4ef)
---
 gdbserver_tests/filter_gdb.in  | 3 +++
 gdbserver_tests/filter_vgdb.in | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/gdbserver_tests/filter_gdb.in b/gdbserver_tests/filter_gdb.in
index 497171732b25..f0400fd18be6 100755
--- a/gdbserver_tests/filter_gdb.in
+++ b/gdbserver_tests/filter_gdb.in
@@ -51,6 +51,9 @@ s/^\[?1034hReading symbols/Reading symbols/
 #       Remove some Suse package manager messages
 /^Try: zypper install -C/d
 
+#      Remove python rpm module load warning
+/^Unable to load 'rpm' module.  Please install the python3-rpm package./d
+
 #       Remove vgdb message
 /relaying data between gdb and process/d
 
diff --git a/gdbserver_tests/filter_vgdb.in b/gdbserver_tests/filter_vgdb.in
index 139aea2bcc52..71f1f702537e 100755
--- a/gdbserver_tests/filter_vgdb.in
+++ b/gdbserver_tests/filter_vgdb.in
@@ -47,5 +47,8 @@ $SED -e '/Use.*info auto-load python-scripts.*/d' |
 $SED -e '/warning: Invalid entry in .debug_gdb_scripts section/d' |
 $SED -e "/Use .info auto-load python .REGEXP.. to list them./d" |
 
+# Remove python rpm module load warning
+$SED -e "/^Unable to load 'rpm' module.  Please install the python3-rpm package./d" |
+
 # and filter out any remaining empty lines
 $SED -e '/^$/d'
-- 
2.45.2

